init-tools.sh

```diff
@@ -99,14 +99,30 @@ if [ ! -e $__DOTNET_PATH ]; then
     echo "Installing dotnet cli..."
     __DOTNET_LOCATION="https://dotnetcli.azureedge.net/dotnet/Sdk/${__DOTNET_TOOLS_VERSION}/${__DOTNET_PKG}.tar.gz"
     # curl has HTTPS CA trust-issues less often than wget, so lets try that first.
-    echo "Installing '${__DOTNET_LOCATION}' to '$__DOTNET_PATH/dotnet.tar'" >> $__init_tools_log
-    if command -v curl > /dev/null; then
-        curl --retry 10 -sSL --create-dirs -o $__DOTNET_PATH/dotnet.tar ${__DOTNET_LOCATION}
-    else
-        wget -q -O $__DOTNET_PATH/dotnet.tar ${__DOTNET_LOCATION}
-    fi
-    cd $__DOTNET_PATH
-    tar -xf $__DOTNET_PATH/dotnet.tar
+    __retryCount=5
+    __retryWaitFactor=6 
+    __count=0
+    until [ $__count -ge $__retryCount ]
+    do
+        echo "Installing '${__DOTNET_LOCATION}' to '$__DOTNET_PATH/dotnet.tar'" >> $__init_tools_log
+        if command -v curl > /dev/null; then
+            curl --retry 10 -sSL --create-dirs -o $__DOTNET_PATH/dotnet.tar ${__DOTNET_LOCATION}
+        else
+            wget -q -O $__DOTNET_PATH/dotnet.tar ${__DOTNET_LOCATION}
+        fi
+        cd $__DOTNET_PATH
+        if tar -xf $__DOTNET_PATH/dotnet.tar; then break; fi
```

Comment:
Is there any cleanup that needs to happen if this fails?

Reply:
Yes.  I added code to clean the directory on failure.